lib/tasks/accounts.rake

```diff
@@ -26,6 +26,8 @@ namespace :accounts do
       puts "Errors in the archive found:\n#{exception.message}\n-----"
     rescue MigrationService::MigrationAlreadyExists
       puts "Migration record already exists for the user, can't continue"
+    ensure
+      service.remove_intermediate_file
```

Comment:
You can just write `ensure` in the `MigrationService` in `perform!` (above the line of calling `remove_intermediate_file` there), that way `remove_intermediate_file` doesn't need to be called twice (and it also ensures that the temp files get cleaned up when `perform!` is called from somewhere else). It also means `remove_intermediate_file` can be private since it doesn't need to be called from external.

Reply:
@SuperTux88 Thanks for reviewing this. In context of #8274 this rake file was more refactored to eliminate duplicate code. So I think for now this file can be as it is.

Reply:
This was more a request to also change `MigrationService.perform!` and add the `ensure` directly there (and when `remove_intermediate_file` doesn't need to be called here anymore it can also be made private).
```ruby
  def perform!
    find_or_create_user
    import_archive
    run_migration
  ensure
    remove_intermediate_file
  end
```